code/modules/power/apc/apc_main.dm

```diff
@@ -596,6 +594,12 @@
 		hacked_flicker_counter = hacked_flicker_counter - 1
 		if(hacked_flicker_counter <= 0)
 			flicker_hacked_icon()
+		if(last_malf_pt_generation + 30 SECONDS < world.time && cell.charge>1000) // Over time generation of malf points for the ai controlling it, costs a bit of power
```

Comment:
Use the cooldown macros 
`COOLDOWN_DECLARE(malf_point_generation`
`COOLDOWN_START(src, malf_point_generation, 30 SECONDS)`
`if(COOLDOWN_FINISHED(src, malf_point_generation)`